core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java

```diff
@@ -50,7 +50,7 @@ public void build(CombinedIndexBuildItem combinedIndexBuildItem, Capabilities ca
 
             boolean methods = getBooleanValue(i, "methods");
             boolean fields = getBooleanValue(i, "fields");
-            boolean ignoreNested = getBooleanValue(i, "ignoreNested");
+            boolean ignoreNested = i.value("ignoreNested") != null && i.value("ignoreNested").asBoolean();
```

Comment:
The original definition of `getBooleanValue` is:

```
private static boolean getBooleanValue(AnnotationInstance i, String name) {
        return i.value(name) == null || i.value(name).asBoolean();
    }
```

which means that it returns `true` if the value is not explicitly set, no matter what the default value is set to in the corresponding annotation, which seems wrong to me.

With the change you propose you revert this and set the default value to false, but it again doesn't seem right to me because:

1. If we change the `ignoreNested` default sometime in the future it will fail again.
2. If we change any other default from true to false it will fail again.

I will prepare a PR to fix this independently.

Reply:
yes, I was looking to a more robust change, to handle properly the default value of the annotation. This commit was to trigger the CI , I have some limitation to test the native version locally. We can do this in a second PR if you prefer ?

Reply:
Yes, let's keep them separate.

Reply:
One thing if we want to do this properly we need to change the getBooleanValue method and it will impact a lot of more test as it will impact the other annotation attributes. Not sure I have sufficient knowledge on the impact of those attributes.

Reply:
@humcqc no worries, I will try to fix this (I already have a WIP branch but hit some issue I am not sure how to resolve yet).

Reply:
@humcqc FYI https://github.com/quarkusio/quarkus/pull/38701